### PR TITLE
Fixing cluster autostop

### PR DESCRIPTION
**Pull request by carolineechen**
creating background thread to periodically (30s) check whether to update last activity time, if within 5 min of autostop.

### CODE REVIEW

**Comment by carolineechen** [comment]
* **#672** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/672?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @carolineechen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/servers/cluster_servlet.py [hunk 1]
```diff
@@ -27,6 +29,34 @@ def __init__(
...
+        from sky.skylet import configs as sky_configs
+
+        while True:
+            autostop = self.cluster_config.get("autostop_mins", -1)
+            self._last_register = sky_configs.get_config("autostop_last_active_time")
+            if autostop > 0 and (
+                not self._last_register
+                or (
+                    # within 5 min of autostop and there's more recent activity
+                    autostop - (time.time() - self._last_register) / 60 < 5
```
Comment (carolineechen) [inline runhouse/servers/cluster_servlet.py:51]: @dongreenberg now that its running in the background do you think it's still useful to check within x min of autostopping or just updating every time last_activity > last_register

Comment (dongreenberg) [reply runhouse/servers/cluster_servlet.py:51]: I think what you have is fine, we can probably make it just 2 minutes though, not 5.

**Review by dongreenberg — APPROVED** [review]
Looks great, TY!

**Pull request merged**